tests/degradation_test_v2.py

```diff
@@ -0,0 +1,175 @@
+            N = len(x)
+            months = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/12, months)
+        elif (input_freq == 'W'):
+            x = pd.date_range(start='2012-01-01', end='2012-01-21', freq=input_freq)
+            N = len(x)
+            weeks = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/52, weeks)
+        elif (input_freq == 'D'):
+            x = pd.date_range(start='2012-01-01', end='2012-01-03', freq=input_freq)
+            N = len(x)
+            days = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/365, days)
+        elif (input_freq == 'H'):
+            x = pd.date_range(start='2012-01-01 00:00:00', end='2012-01-02 00:00:00', freq=input_freq)
+            N = len(x)
+            hours = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/(365 * 24), hours)
+        elif (input_freq == 'T'):
+            x = pd.date_range(start='2012-01-01 00:00:00', end='2012-01-01 00:50:00', freq=input_freq)
+            N = len(x)
+            minutes = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/(365 * 24 * 60), minutes)
+        elif (input_freq == 'S'):
+            x = pd.date_range(start='2012-01-01 00:00:00', end='2012-01-01 00:10:00', freq=input_freq)
+            N = len(x)
+            seconds = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/(365 * 24 * 60 * 60), seconds)
+        else :
+            sys.exit("Unknown frequency type")
+        
+        corr_energy = pd.Series(data=y, index=x)
+        return corr_energy
+
+    def get_CD_YOY_corr_energy(self, rd, input_freq):
+        ''' Create input for degradation_classical_decomposition and degradation_year_on_year
+            functions, depending on frequency. Allowed frequencies for both of these degradation 
+            functions  are 'MS', 'M', 'W' and 'D'.
+        '''
+        if (input_freq == 'MS'):
+            x = pd.date_range(start='2012-01-01', end='2015-01-01', freq=input_freq)
+            N = len(x)
+            months = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/12, months)
+        elif (input_freq == 'M'):
+            x = pd.date_range(start='2012-01-31', end='2015-01-31', freq=input_freq)
+            N = len(x)
+            months = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/12, months)
+        elif (input_freq == 'W'):
+            x = pd.date_range(start='2012-01-01', end='2015-01-01', freq=input_freq)
+            N = len(x)
+            weeks = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/52, weeks)
+        elif (input_freq == 'D'):
+            x = pd.date_range(start='2012-01-01', end='2015-01-01', freq=input_freq)
+            N = len(x)
+            days = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/365, days)
+        else :
+            sys.exit("Unknown frequency type")
+        
+        corr_energy = pd.Series(data=y, index=x)
+        return corr_energy
+
+
+    def setUp(self):
```

Comment:
The setUp method runs before every test function (functions named test_). So this method ends up running three times. We only need to define the test data once, so what we need is [setUpClass](https://docs.python.org/2/library/unittest.html#unittest.TestCase.setUpClass).

All you need to do is rename `setUp(self)` to `setUpClass(cls)`, add the `@classmethod` decorator, and replace `self` with `cls`.